lizard_auth_server/models.py

```diff
@@ -27,8 +34,41 @@
 logger = logging.getLogger(__name__)
 
 
+from django.utils.deconstruct import deconstructible
+
+
+@deconstructible
+class GenKey(object):
+    # Field.default needs to be serializable (a change since Django 1.7). Here
+    # we use the deconstruct method described in:
+    # https://code.djangoproject.com/ticket/22999
+    # https://docs.djangoproject.com/en/1.9/topics/migrations/#adding-a-deconstruct-method
+
+    def __init__(self, model, field):
+        self.model = model
+        self.field = field
+
+    def __call__(self):
+        if isinstance(self.model, basestring):
+            ModelClass = get_model('lizard_auth_server', self.model)
+            if not ModelClass:
+                raise Exception('Unknown model {}'.format(self.model))
+        else:
+            ModelClass = self.model
+        key = gen_secret_key(64)
+        while ModelClass.objects.filter(**{self.field: key}).exists():
+            key = gen_secret_key(64)
+        return key
+
+    def __eq__(self, other):
+        return all([self.model == other.model,
+                    self.field == other.field])
+
+
 def gen_key(model, field):
     """
+    DEPRECATED, reimplemented in GenKey which is serialiazable.
```

Comment:
Deze functie mag dan ook gewoon weg, we zijn de enige die onze eigen api gebruiken.